### PR TITLE
feat: Implement domain name validation on client and server

### LINKED ISSUE #589: Enhance Domain Input Validation in Team Settings
### Guidelines 👇🏼
If you would like to be rewarded by BoxyHQ for your participation in contributing to Hacktoberfest, [please register for our Hacktoberfest event](https://events.mlh.io/events/10461-boxyhq-hacktoberfest-2023).

___

**Description:**
In the Team Settings, the domain input field currently accepts any string as input, which can lead to inaccuracies and inconsistencies. To improve data integrity and user experience, we should implement domain name validation both in the frontend and backend.

**Expected Behavior:**
When users enter a domain in the **Team Settings**, the system should verify that it's a valid domain name format. This validation should occur in real-time on the frontend to provide immediate feedback and should also be enforced on the backend to ensure data consistency.

**Proposed Solution:**
- Implement frontend validation that checks domain name format as users type.
- Enforce backend validation to reject invalid domain names when saving settings.
- Provide clear error messages for users attempting to enter an invalid domain name.

**Additional Information:**
Implementing this validation will ensure that domain names entered in Team Settings are accurate and valid, reducing potential errors and ensuring data consistency across the platform.

### PR DESCRIPTION
**Pull request by Dhoni77**
Closes #589 

- Implement domain name validation on client and server using regex.
- Regex taken from [https://regexr.com/3au3g](url)
- Included error messages

### CODE REVIEW

File: pages/api/teams/index.ts
Comment (devkiran) [inline pages/api/teams/index.ts:1]: This check should do while update team info, not creating. 
https://github.com/boxyhq/saas-starter-kit/blob/main/pages/api/teams/%5Bslug%5D/index.ts#L55

**Review by devkiran — CHANGES_REQUESTED** [review]

File: pages/api/teams/index.ts
Comment (devkiran) [reply pages/api/teams/index.ts:1]: @Dhoni77 Please check the changes requested.

Comment (Dhoni77) [reply pages/api/teams/index.ts:1]: done

File: pages/api/teams/[slug]/index.ts [hunk 1]
```diff
@@ -57,10 +59,16 @@ const handlePUT = async (req: NextApiRequest, res: NextApiResponse) => {
   const teamMember = await throwIfNoTeamAccess(req, res);
   throwIfNotAllowed(teamMember, 'team', 'update');
 
+  const { name, slug, domain } = req.body;
+
+  if (!validateDomain(domain)) {
```
Comment (devkiran) [inline pages/api/teams/[slug]/index.ts:64]: @Dhoni77 `domain` should be optional. That means validate it only if it is not empty.

**Review by devkiran — CHANGES_REQUESTED** [review]

File: pages/api/teams/[slug]/index.ts (hunk 1 above)
Comment (Dhoni77) [reply pages/api/teams/[slug]/index.ts]: done

**Review by devkiran — APPROVED** [review]

**Pull request merged**